src/Language/JVM/CFG.hs

```diff
@@ -419,7 +419,7 @@ retTargetXfer extbl istrm acc (Just pc, localr) = xfer (lkup pc)
 
 leaderPC :: BasicBlock -> PC
 leaderPC BB{ bbInsts = [] } = error "internal: leaderPC on empty BB"
-leaderPC bb                 = fst . head . bbInsts $ bb
+leaderPC bb                 = headInsts . bbInsts $ bb
```

Comment:
The `headInsts` function already has protection against empty lists, so line 421 is no longer needed, unless you prefer the separate error message. 

It's unclear to me whether any call to `headInts` (here or elsewhere) would ever be expected to get a null array, or if this is equivalent to panic.  If it's the latter, long term (i.e. in a future PR, not necessarily this one) I'd love to see us use the actual `Panic.panic` function for these; if it's the former, maybe the callers should pass a self-identifying string argument that could be printed by `headInts` along with the existing error message.  Either way, I'd vote for removing line 421 here.

Reply:
Good catch. I've removed line 421.